### PR TITLE
changelog: mention db-user and db-name changes for Teleport 14

### PR DESCRIPTION
**Pull request by GavinFrazar**
This PR documents a behavioral change in `tsh db` subcommands in the v14 changelog. Relevant PR introducing the change: https://github.com/gravitational/teleport/pull/26564

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,28 @@
 # Changelog
 
+## 14.0.0 (xx/xx/23)
+
+### Breaking Changes
+
+Please familiarize yourself with the following potentially disruptive changes
+in Teleport 14 before upgrading.
+
+#### `tsh db` --db-user and --db-name defaults
```
Comment (alexfornuto) [inline CHANGELOG.md:10]:
```suggestion
#### `tsh db --db-user` and `--db-name` defaults
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+
+In Teleport 14 `tsh db` sub-commands will check if `--db-user` or `--db-name`
+flags are required and not provided by the user, and attempt to choose a default
+value.
+If the user's role(s) matching the database's labels have the `create_db_user`
+option enabled, then the Teleport user's username will be chosen as the default
+for `--db-user`.
+Otherwise, a default is chosen by examining the allowed `db_users` and
+`db_names` in the user's role(s) that apply for the given database.
+If, after applying user traits, only one non-wildcard db name/user is allowed,
```
Comment (alexfornuto) [inline CHANGELOG.md:20]: Maybe not here, but we should break up the text block somewhere for easier readability.

```suggestion

If, after applying user traits, only one non-wildcard db name/user is allowed,
```

File: CHANGELOG.md (hunk 1 above)
Comment (GavinFrazar) [reply CHANGELOG.md:10]: What is the intent of this suggestion? Would this be better / more clear:
`tsh [proxy] db [login | connect]` flags `--db-user` and `--db-name`

File: CHANGELOG.md (hunk 2 above)
Comment (GavinFrazar) [reply CHANGELOG.md:21]: added

**Review by r0mant — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (GavinFrazar) [reply CHANGELOG.md:10]: @alexfornuto

**Review by smallinsky — APPROVED** [review]

**Pull request merged**